airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/repository.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any, Iterable, Mapping, Set
+
+from airbyte_cdk import StreamSlice
+from airbyte_cdk.sources.declarative.async_job.job import AsyncJob
+
+
+@dataclass
```

Comment:
why is the interface a dataclass?

Reply:
I'll remove that. I guess this was copied from other instances where we were doing this like [here](https://github.com/airbytehq/airbyte/pull/45178/files/07b24c2971c0190ca2462f3b2f1424f132e200bb#diff-74b0a729c5ee94162418441cea80512a9d27920954ed1448547d0334ca2f62aeL14)